packages/expo-asset/ios/AssetModule.swift

```diff
@@ -0,0 +1,88 @@
+import ExpoModulesCore
+import CryptoKit
+
+internal class UnableToDownloadAssetException: GenericException<URL> {
+  override var reason: String {
+    "Unable to download asset from url: \(param)"
+  }
+}
+
+public class AssetModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoAsset")
+
+    AsyncFunction("downloadAsync") { (url: URL, md5Hash: String?, type: String, promise: Promise) in
+      if url.isFileURL {
+        promise.resolve(url.standardizedFileURL.absoluteString)
+      }
+      guard let cacheFileId = md5Hash ?? getMD5Hash(fromURL: url),
+      let cachesDirectory = appContext?.fileSystem?.cachesDirectory,
+      let appContext = appContext else {
+        promise.reject(UnableToDownloadAssetException(url))
+        return
+      }
+
+      let localUrl = URL(fileURLWithPath: "\(cachesDirectory)/ExponentAsset-\(cacheFileId).\(type)")
+
+      guard let fileData = FileManager.default.contents(atPath: localUrl.path) else {
+        downloadAsset(appContext: appContext, url: url, localUrl: localUrl, promise: promise)
+        return
+      }
+      if md5Hash == nil {
+        promise.resolve(localUrl.standardizedFileURL.absoluteString)
+        return
+      }
+      if md5Hash == getMD5Hash(fromData: fileData) {
+        promise.resolve(localUrl.standardizedFileURL.absoluteString)
+        return
+      }
+      downloadAsset(appContext: appContext, url: url, localUrl: localUrl, promise: promise)
+    }
+  }
+
+  private func getMD5Hash(fromURL url: URL) -> String? {
+    guard let urlData = url.absoluteString.data(using: .utf8) else {
+      return nil
+    }
+    return getMD5Hash(fromData: urlData)
+  }
+
+  private func getMD5Hash(fromData data: Data?) -> String? {
+    guard let data = data else {
+      return nil
+    }
+    return Data(Insecure.MD5.hash(data: data)).base64EncodedString()
```

Comment:
at a glance it seems that android is using hex encoding while iOS is using base64?